cesium/internal/domain/writer.go

```diff
@@ -259,9 +301,9 @@ func (w *Writer) Close(ctx context.Context) error {
 	return w.internal.Close()
 }
 
-func (w *Writer) validateCommitRange(end telem.TimeStamp) error {
-	if !w.prevCommit.IsZero() && end.Before(w.prevCommit) {
-		return errors.Wrap(validate.Error, "commit timestamp must be strictly greater than the previous commit")
+func (w *Writer) validateCommitRange(end telem.TimeStamp, switchingFile bool) error {
+	if !w.prevCommit.IsZero() && !switchingFile && end.Before(w.prevCommit) {
```

Comment:
Can you explain why we don't validate that the end was after the previous commit when switching files?

Reply:
If there is a preset end, the previous commit will take that as prevCommit, but we must terminate the commit prematurely in order to open up a new one. therefore.

Reply:
Changed to `&& !(switchingFile && w.presetEnd)`